OnlineShopUI/online-shop-ui/src/components/ModalLogin/Login.js

```diff
@@ -43,25 +51,60 @@ export default function Login() {
                 onClick={handleModalClose}
               ></CloseIcon>
             </div>
-            <div className="login__date">
+
+            <form className="login__date" onSubmit={handleSubmit(onSubmit)}>
               <div className="login__email">
                 <label>Username</label>
                 <input
                   type="text"
                   onChange={(e) => {
                     setUsername(e.target.value);
                   }}
+                  {...register('Username', {
+                    required: <p>Field Username required for registration</p>,
+                    minLength: {
+                      value: 3,
+                      message: <p>Username must be more than 3 characters</p>,
+                    },
+                  })}
                 />
               </div>
+              <div className="login__error">
+                {errors?.Username && (
+                  <p>
+                    <img src={Error} alt="Erorr" />
+                    {errors?.Username.message || 'Error!'}
+                  </p>
+                )}
+              </div>
               <div className="login__pass">
                 <label>Пароль</label>
                 <input
                   type="password"
                   onChange={(e) => {
                     setPassword(e.target.value);
                   }}
+                  {...register('Password', {
+                    required: 'Field Password required for registration',
+                    pattern: {
+                      value:
+                        /^(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{5,}$/,
+                    },
+                    minLength: {
+                      message: 'Password must be more than 5 characters',
+                    },
```

Comment:
Looks like your regexp already covers the situation when password less than 5, or I'm wrong?

Reply:
Yes, I agree with you!